micrometer-core/src/main/java/io/micrometer/core/instrument/observation/ObservationOrTimerCompatibleInstrumentation.java

```diff
@@ -121,6 +124,15 @@ public <RES> void setResponse(RES response) {
         responseContext.setResponse(response);
     }
 
+    /**
+     * If using an Observation will set the error on Observation. For metrics it will set
+     * a tag {@code error}.
+     * @param throwable error that got recorded
+     */
+    public void setThrowable(Throwable throwable) {
```

Comment:
@shakuzen I'm setting throwable here and `error` tag for metrics if there was an error (or `none` value if there was none)

Reply:
I'm not sure if we should be doing that for Timer. That will add an `error` tag where there maybe wasn't one before. An additional tag shouldn't be a problem most of the time when querying, but it can depend how the query is written. Sometime Prometheus queries are written using a [`without` clause](https://prometheus.io/docs/prometheus/latest/querying/operators/#aggregation-operators).

In addition to that, some instrumentation may be tagging an error on Timer in a different way - for example, existing Spring http-related instrumentation uses `exception` rather than `error`.

Another interesting side effect is verifying the instrumentation matches the DocumentedObservation. If we are strictly checking there aren't tags/keyvalues produced that aren't documented, then people need to add `error` to their DocumentedObservation low cardinality keyvalues.